kratos/solving_strategies/schemes/scheme.h

```diff
@@ -502,6 +502,39 @@ class Scheme
         )
     {
         KRATOS_TRY
+
+        ProcessInfo& r_current_process_info = rModelPart.GetProcessInfo();
+
+        // Definition of the first element iterator
+        const auto it_elem_begin = rModelPart.ElementsBegin();
+
+        // Initialize non-linear iteration for all of the elements
+        #pragma omp parallel for
+        for(int i=0; i<static_cast<int>(rModelPart.Elements().size()); ++i) {
+            auto it_elem = it_elem_begin + i;
+            it_elem->InitializeNonLinearIteration(r_current_process_info);
+        }
+
+        // Definition of the first condition iterator
+        const auto it_cond_begin = rModelPart.ConditionsBegin();
+
+        // Initialize non-linear iteration for all of the conditions
+        #pragma omp parallel for
+        for(int i=0; i<static_cast<int>(rModelPart.Conditions().size()); ++i) {
+            auto it_cond = it_cond_begin + i;
+            it_cond->InitializeNonLinearIteration(r_current_process_info);
+        }
+
+        // Definition of the first constraint iterator
+        const auto it_const_begin = rModelPart.MasterSlaveConstraintsBegin();
+
+        // Initialize non-linear iteration for all of the constraints
+        #pragma omp parallel for
+        for(int i=0; i<static_cast<int>(rModelPart.MasterSlaveConstraints().size()); ++i) {
```

Comment:
@loumalouomega

Reply:
i think that this was deliberately not called here but rather directly within the builder and solver.

Reply:
nope, the BuilderAndSolver never did this, I checked
Actually none did this I am quite sure